File: Ruby/problem04.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

def problem04(list)
  list.map {|e|
    e.to_s
  }.sort {|a, b|
    b <=> a
  }.join.to_i
end


if __FILE__ == $0
  p problem04([50, 2, 1, 9])
end

File: JavaScript/problem04.js
;(function(print) {
  'use strict';

  function problem04(list) {
    return +list.map(function(e) {
      return e + '';
    }).sort().reverse().join('');
  }

  print(problem04([50, 2, 1, 9]));
})(typeof console === 'undefined' ? print : console.log);
